logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreFailure.kt

```diff
@@ -90,6 +95,24 @@ internal inline fun <T : Any> wrapCryptoRequest(cryptoRequest: () -> T): Either<
     }
 }
 
+internal inline fun <T : Any> wrapMLSRequest(mlsRequest: () -> T): Either<MLSFailure, T> {
+    return try {
+        Either.Right(mlsRequest())
+    } catch (e: Exception) {
+        kaliumLogger.e(e.stackTraceToString())
+        Either.Left(MLSFailure(e))
+    }
+}
+
+internal inline fun <T : Any> wrapMLSNullableRequest(mlsRequest: () -> T?): Either<MLSFailure, T?> {
+    return try {
+        Either.Right(mlsRequest())
+    } catch (e: Exception) {
+        kaliumLogger.e(e.stackTraceToString())
+        Either.Left(MLSFailure(e))
+    }
+}
```

Comment:
Instead of having a nullable and a normal request, we can have a single one:

```kotlin
internal inline fun <T> wrapMLSRequest(mlsRequest: () -> T): Either<MLSFailure, T> {
    return try {
        Either.Right(mlsRequest())
    } catch (e: Exception) {
        kaliumLogger.e(e.stackTraceToString())
        Either.Left(MLSFailure(e))
    }
}
```

The difference is that it doesn't  have `T: Any`. So T can be any nullable type.

Reply:
Also, can we have a simple test for this one?

```kotlin
@Test 
fun givenExceptionIsThrown_whenWrappingMLSRequest_thenShouldReturnMLSFailureWithCorrectCause() {
    val exception = IllegalArgumentException("Test exception")

    val result = wrapMLSRequest {
        throw exception
    }
    
    result.shouldFail {
        assertEquals(exception, it)
    }
}

```

Reply:
Done